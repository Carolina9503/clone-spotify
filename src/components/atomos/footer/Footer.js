import React from "react";
import styles from "./Footer.module.scss";
import ShuffleIcon from "@material-ui/icons/Shuffle";
import SkipPreviousIcon from "@material-ui/icons/SkipPrevious";
import PlayCircleOutlineIcon from "@material-ui/icons/PlayCircleOutline";
import SkipNextIcon from "@material-ui/icons/SkipNext";
import RepeatIcon from "@material-ui/icons/Repeat";
import { Slider, Grid } from "@material-ui/core";
import PlaylistPlayIcon from "@material-ui/icons/PlaylistPlay";
import VolumeDownIcon from "@material-ui/icons/VolumeDown";

const Footer = () => {
  return (
    <div className={styles.footer}>
      <div className={styles.footer__left}>
        <img
          src="https://i.scdn.co/image/ab67616d00004851e6f407c7f3a0ec98845e4431"
          alt="covers"
        />
        <div>
          <h4>Off My Face</h4>
          <p>Justin Bieber</p>
        </div>
      </div>
      <div className={styles.footer__center}>
        <ShuffleIcon className={styles.shuffle} />
        <SkipPreviousIcon className={styles.icon} />
        <PlayCircleOutlineIcon className={styles.icon} />
        <SkipNextIcon className={styles.icon} />
        <RepeatIcon className={styles.repeat} />
      </div>
      <div className={styles.footer__right}>
        <Grid container spacing={3} className={styles.grid_footer}>
          <Grid item>
            <PlaylistPlayIcon className={styles.icon} />
          </Grid>
          <Grid item>
            <VolumeDownIcon className={styles.icon} />
          </Grid>
          <Grid item xs>
            <Slider className={styles.MuiSlider_root} />
          </Grid>
        </Grid>
        <div></div>
      </div>
    </div>
  );
};

export default Footer;
