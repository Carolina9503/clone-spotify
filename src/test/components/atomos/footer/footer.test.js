import { shallow } from "enzyme";
import Footer from "../../../../components/atomos/footer/Footer";

describe("Pruebas en <Footer/>", () => {
  const wrapper = shallow(<Footer />);

  test("debe de mostarr el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe tener una imagen con una url y un alt", () => {
    const img = wrapper.find("img");
    const url =
      "https://i.scdn.co/image/ab67616d00004851e6f407c7f3a0ec98845e4431";
    const alt = "covers";

    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(alt);
  });

  test("debe tener un div con los iconos de la barra", () => {
    const div = wrapper.find("div");
    console.log(wrapper.contains());
  });
});
