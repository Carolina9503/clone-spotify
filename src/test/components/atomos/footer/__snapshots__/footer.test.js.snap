// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Footer/> debe de mostarr el componente correctamente 1`] = `
<div
  className="footer"
>
  <div
    className="footer__left"
  >
    <img
      alt="covers"
      src="https://i.scdn.co/image/ab67616d00004851e6f407c7f3a0ec98845e4431"
    />
    <div>
      <h4>
        Off My Face
      </h4>
      <p>
        Justin Bieber
      </p>
    </div>
  </div>
  <div
    className="footer__center"
  >
    <Memo(ForwardRef(ShuffleIcon))
      className="shuffle"
    />
    <Memo(ForwardRef(SkipPreviousIcon))
      className="icon"
    />
    <Memo(ForwardRef(PlayCircleOutlineIcon))
      className="icon"
    />
    <Memo(ForwardRef(SkipNextIcon))
      className="icon"
    />
    <Memo(ForwardRef(RepeatIcon))
      className="repeat"
    />
  </div>
  <div
    className="footer__right"
  >
    <WithStyles(ForwardRef(Grid))
      className="grid_footer"
      container={true}
      spacing={3}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <Memo(ForwardRef(PlaylistPlayIcon))
          className="icon"
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <Memo(ForwardRef(VolumeDownIcon))
          className="icon"
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        className="slider"
        item={true}
        xs={true}
      >
        <WithStyles(ForwardRef(Slider))
          className="MuiSlider_root"
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <div />
  </div>
</div>
`;
